src/ensembl/src/content/app/entity-viewer/gene-view/GeneView.tsx

```diff
@@ -133,10 +141,23 @@ const GeneViewWithData = (props: GeneViewWithDataProps) => {
     setBasePairsRulerTicks
   ] = useState<TicksAndScale | null>(null);
 
+  const history = useHistory();
   const params: { [key: string]: string } = useParams();
   const { genomeId, entityId } = params;
   const gbUrl = urlFor.browser({ genomeId, focus: entityId });
 
+  useEffect(() => {
+    if (props.entityViewerQueryParams.view) {
+      props.setGeneViewMode(props.entityViewerQueryParams.view);
+    }
+  }, [props.entityViewerQueryParams.view]);
+
+  const changeViewMode = (tab?: string) => {
+    const view = getViewModeName(props.selectedGeneTabName, tab);
+    const newPath = urlFor.entityViewer({ genomeId, entityId, view });
+    history.push(newPath);
```

Comment:
We usually do this through connected react router.

Reply:
✔️